### ISSUE TITLE
Collection shapes

### ISSUE THREAD

**Issue by tpetry** [nunomaduro/larastan#880]
- Larastan Version: 0.7.10
- `--level` used: 8

### Description

Is there some way to specify collection shapes which are enforced like array shapes in phpstan? The type hints for arrays do work, but not for collections.

### Code where the issue was found

```php
<?php

declare(strict_types=1);

namespace App\Sketch;

use Illuminate\Support\Collection;

class Aaa
{
    /**
     * @return \Illuminate\Support\Collection<string>
     */
    public function collection(): Collection
    {
        return collect([88, 99]); // should be invalid
    }

    /**
     * @return array<string>
     */
    public function array(): array
    {
        return [88, 99]; // is invalid
    }
}
```

```
 ------ ----------------------------------------------------------------------------------------------------------------
  Line   Aaa.php
 ------ ----------------------------------------------------------------------------------------------------------------
  24     Method App\Sketch\Aaa::array() should return array<string> but returns array<int, int>.
 ------ ----------------------------------------------------------------------------------------------------------------
```

**Comment by tpetry** [comment]
I was now playing a lot more with collections and transforming them to arrays to get type checks and as soon as I use collections the type checking does not work anymore. Am i doing something wrong, or can collections just not be type checked?

```php
<?php

declare(strict_types=1);

namespace App\Sketch;

class Bbb
{
    /**
     * @return array<array{id: string}>
     */
    public function collection(): array
    {
        return collect([4,5,6])->map(fn ($x) => ['id' => $x])->all(); // should be invalid
    }

    /**
     * @return array<array{id: string}>
     */
    public function array(): array
    {
        return array_map(fn ($x) => ['id' => $x], [4,5,6]); // is invalid
    }
}
```

```
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  Line   Bbb.php
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  22     Method App\Sketch\Bbb::array() should return array<array('id' => string)> but returns array(array('id' => 4|5|6), array('id' => 4|5|6), array('id' => 4|5|6)).
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

```

**Comment by canvural** [comment]
Hello,

Looks like this is not supported yet. There needs to be a new extension and/or changes to stub files. So I'll mark it as a feature request.

**Comment by canvural** [comment]
Implemented in #904
